src/main/java/com/purbon/kafka/topology/Configuration.java

```diff
@@ -374,11 +378,40 @@ public String getKafkaConnectClusterId() {
         .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
   }
 
-  public String getKSQLServer() {
-    return config.getString(PLATFORM_SERVER_KSQL);
+  public URL getKSQLServer() {
+    if (config.hasPath(PLATFORM_SERVER_KSQL_URL)) {
+      try {
+        return new URL(config.getString(PLATFORM_SERVER_KSQL_URL));
+      } catch (MalformedURLException e) {
+        throw new ConfigurationException(config.getString(PLATFORM_SERVER_KSQL_URL) + " is not a valid KSQL URL");
+      }
+    }
+    return null;
+  }
+
+  public BasicAuth getKSQLBasicAuth() {
+    return new BasicAuth(config.getString(PLATFORM_SERVER_KSQL_BASIC_AUTH_USER), config.getString(PLATFORM_SERVER_KSQL_BASIC_AUTH_PASSWORD));
   }
 
   public boolean hasKSQLServer() {
-    return config.hasPath(PLATFORM_SERVER_KSQL);
+    return config.hasPath(PLATFORM_SERVER_KSQL_URL);
+  }
+
+  public static class BasicAuth {
+    private final String user;
+    private final String password;
+
+    public BasicAuth(String user, String password) {
+      this.user = user;
+      this.password = password;
+    }
+
+    public String getUser() {
+      return user;
+    }
+
+    public String getPassword() {
+      return password;
+    }
```

Comment:
should be interesting to add a new function to calculate the base64 file for generating a basic auth token?

Reply:
I'll extract the base64 encoding which is currently present in the JulieHttpClient to the BasicAuth class